README.md

```diff
@@ -1,40 +1,88 @@
-# `ceasar-colyseus`
-Based on `create-colyseus-app`
+# `ceasar-server`
+
+A CEASER Game server based on `create-colyseus-app`.
 
 ## :crossed_swords: Usage
-If this is the first time running the app, run `npm install`
+* If this is the first time running the app, run: `npm install`
+* If things are busted, try a nuke-and-repave: `./repave.sh`
+
+## Development:
+
+### Local requirements:  
+You will need to add a local `.env` file to this project directory.
+*Important: `.env` should always be included in `.gitignore`*
 
-Start server locally:
+The `.env` file should contain one environmental variable for the mongo server:
 
-`npm start`
+```
+    MONGO_URI=mongodb://[user]:[password]@[host]
+```
 
-Rebuild ready for deploy on Heroku:
+You will need to obtain that URI from another developer or from
+Config Var of the same name in the Heroku [ceaser-server-staging app](https://dashboard.heroku.com/apps/ceaser-server-staging/settings).
 
-`tsc`
+### Running Locally:
 
-When pushing to Heroku, the contents of the dist folder are served up.
+To start a live-reloading development server run:
+```
+   npm run live
+```
+
+Your Ceaser Websocket should then be available at `ws://localhost:2567`.
+
+## Changing the Schema:
 
 To regenerate the schema for the room, assuming project CEASAR is at same level as this on your system:
 
-`npx schema-codegen CeasarRoom.ts --csharp --output ../CEASAR/Assets/Scripts/Network/`
+```
+   npx schema-codegen CeasarRoom.ts --csharp --output ../CEASAR/Assets/Scripts/Network/
+```
+
+## Heroku Deployment:
+Ceaser-server is Heroku app setup under the Concord Consortium account.
+
+Heroku will look for pushed github branches to run as apps. *`master`* is automatically
+deployed. Other branches can be configured using the Heroku web console as preview apps.
+
+The Heroku pipeline looks at `package.json` to figure out how to build and run
+the Caeser-server app. First `heroku-postbuild` is run which builds the `/dist`
+folder, then `start` is run, which executes `node dist/index.ts`.
+
+### Preview Branches
+Pushing changes to *any* branch should cause a Heroku preview app to be built.
+You will need to configure the apps settings, to add a `MONGO_URI` config var.
+For now you can just copy the `MONGO_URI` config var from the
+[staging-app settings](https://dashboard.heroku.com/apps/ceasar-server-staging/settings).
+
+### Staging Branch
+Pushing changes to *`master`* will cause the Heroku staging sever to update.
+THe staing server is already provisioned with configured MLAb Mongo Heroku addon.
+
+The heroku app websocket server should be available at:
+wss://ceasar-server-staging.concord.org/ or
+
+### Procution Branch
```

Comment:
Production? - think you need a new keyboard? ;)

Reply:
Unfortunately I can't blame this one on my keyboard -- the "d" key works and is where its supposed to be.  Its my fingers / brain that are broken.  :)

(All the extra space characters and U characters I can blame on this laptop though ... )